src/scripts/branch_review.py

```diff
@@ -0,0 +1,312 @@
+                        curie=child,
+                        relationships=children_of_obsoletion_candidate_parents_relationship,
+                    )
+                )
+                - all_obsoletes
+            )
+
+            # These are parents that need to be obsoleted and hence NOT in Branch
+            relevant_obsoletion_parents_in_branch = (
+                parents_of_child & obsoletion_candidate_child_of_branch
+            )
+            # if child == "MONDO:0013310":
+            #     import pdb; pdb.set_trace()
+
+            # parents_of_child.difference_update(obsoletion_candidate_child_of_branch)
+
+            # Column F: Other direct parents in Branch = other_parents_in_branch
+            other_parents_in_branch = parents_of_child & branch_descendants_set
+
+            # Column G: Other direct parents NOT in Branch = other_parents_not_in_branch
+            other_parents_not_in_branch = parents_of_child - other_parents_in_branch
+
+            all_ancestors = (
+                set(
+                    OI.ancestors(
+                        [child], predicates=[IS_A, PART_OF], method="ENTAILMENT"
+                    )
+                )
+                # - parents_of_child
+                - {child}
+            )
+            all_ancestors = {
+                anc for anc in all_ancestors if str(anc).startswith("MONDO:")
+            }
+
+            # all_ancestors.difference_update(obsoletion_candidate_child_of_branch)
+
+            # Column H: Ancestor inside the branch = all_mondo_ancestors_in_branch
+            all_ancestors_in_branch = all_ancestors & branch_descendants_set
+
+            # Column I: Ancestor outside the branch = all_mondo_ancestors_outside_branch
+            all_ancestors_outside_branch = all_ancestors - all_ancestors_in_branch
+
+            # if child == "MONDO:0013310":
+            #     import pdb; pdb.set_trace()
+
+            other_parents_in_branch = add_obsoleted_label(
+                other_parents_in_branch, obsoletion_candidates_set
+            )
+            other_parents_not_in_branch = add_obsoleted_label(
+                other_parents_not_in_branch, obsoletion_candidates_set
+            )
+            all_ancestors_in_branch = add_obsoleted_label(
+                all_ancestors_in_branch, obsoletion_candidates_set
+            )
+            all_ancestors_outside_branch = add_obsoleted_label(
+                all_ancestors_outside_branch, obsoletion_candidates_set
+            )
+            relevant_obsoletion_parents_in_branch = add_obsoleted_label(
+                relevant_obsoletion_parents_in_branch, obsoletion_candidates_set
+            )
+
+            # Column J: Affected Status = status
+            status = ""
+            if len(other_parents_in_branch) > 0 and any(
+                " - TO_BE_OBSOLETED" not in parent for parent in other_parents_in_branch
+            ):
+                status = STAYS_IN_THE_BRANCH
```

Comment:
@sabrinatoro in my opinion, this status should be renamed to: "Retains parent in branch" for clarity

Reply:
We probably also want a code that expresses if the class retains an ancestor in he branch, and check that right after this one, so maybe this is a better naming scheme?:

1. Retains parent in branch (currently `STAYS_IN_THE_BRANCH`)
2. Retains ancestor in branch (currently missing)
3. Leaves branch but retains parent in other branch (currently `LEAVES_THE_BRANCH`)
4. Orphaned (leave as is)